tectonic/dpx-cs_type2.c

```diff
@@ -731,7 +731,7 @@ do_charstring (card8 **dest, card8 *limit,
       do_operator2(dest, limit, data, endptr);
     } else if (b0 < 32 && b0 != 28) { /* 19, 20 need mask */
       do_operator1(dest, limit, data, endptr);
-    } else if ((b0 <= 22 && b0 >= 27) || b0 == 31) { /* reserved */
+    } else if ((b0 >= 22 && b0 <= 27) || b0 == 31) { /* reserved */
```

Comment:
Sure, but I think this is not the first time we encounter a real bug in the dpx code caught thanks to a warning or similar (e.g., I remember fixing a use after free). I think I should try to come up with a list and point them to the commits in tectonic to see what they think. I will do that later when I have some time.